mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -1282,6 +1282,13 @@ OpFoldResult arith::MulFOp::fold(FoldAdaptor adaptor) {
   if (matchPattern(adaptor.getRhs(), m_OneFloat()))
     return getLhs();
 
+  if (arith::bitEnumContainsAll(getFastmath(), arith::FastMathFlags::nnan |
+                                                   arith::FastMathFlags::nsz)) {
```

Comment:
doesn't it need also ninf? inf * 0 -> Nan

Reply:
I tried to check this with Alive: https://alive2.llvm.org/ce/z/wvNkdy